FreeBSD_12.1_build_fix_and_config_updates

bobr-kun opened the pull request:
Added fix to prevent build crash under FreeBSD 12.1. Tested on clean FreeBSD 12.1 releases:
    FreeBSD 12.1-RELEASE FreeBSD 12.1-RELEASE
    FreeBSD 12.1-RELEASE-p13 FreeBSD 12.1-RELEASE-p13 GENERIC  amd64
    
Added 'port' option to configs

bobr-kun committed:
Added fix to prevent build crash under FreeBSD 12.1. Tested on clean FreeBSD 12.1 releases:
    FreeBSD 12.1-RELEASE FreeBSD 12.1-RELEASE
    FreeBSD 12.1-RELEASE-p13 FreeBSD 12.1-RELEASE-p13 GENERIC  amd64

Added 'port' option to configs

madf commented on projects/sgconv/sgconv.conf:
```diff
@@ -121,4 +121,9 @@ Store module
+    # Database port
+    # Parameter: optional
+    # Value: any, supported by database
+    # Default: 0
```
Default port shouldn't be zero.

madf replied:
`Default` still says zero.

madf commented on projects/stargazer/inst/freebsd/etc/stargazer/conf-available.d/store_mysql.conf:
```diff
@@ -23,4 +23,10 @@
+    # Parameter: optional
+    # Value: any, supported by database
+    # Default: 0
+    Port = 0
```
Please comment-out this line, the parameter is optional. Besides, 0 is a wrong value.

madf commented on stglibs/pinger.lib/include/stg/pinger.h:
```diff
@@ -19,6 +19,7 @@
 #endif
 
 #if defined (FREE_BSD) || defined (FREE_BSD5) || defined(DARWIN)
+#include <sys/types.h>
```
I don't think this fix applies for Darwin platform. Please limit it to FreeBSD and likely to FreeBSD5 and later only - there was no such problem on FreeBSD4 with gcc.

madf replied:
That's wrong. Stargazer has its own OS detection macros, and they are `FREE_BSD` and `FREE_BSD5`. You need the latter.

bobr-kun committed:
Changed including of the <sys/types.h> for __FreeBSD__ >= 5 only.

Default value of the MySQL 'port' option in configs changed to 3306.

Built and compiled under FreeBSD 12.1 with gcc/g++ and clang/clang++:
    no errors at all with gcc/g++
    same 2 warnings(as always) with clang/clang++

madf commented on projects/stargazer/inst/freebsd/etc/stargazer/conf-available.d/store_mysql.conf:
```diff
@@ -23,4 +23,10 @@
+    # Parameter: optional
+    # Value: any, supported by database
+    # Default: 0
+    Port = 3306
```
Please comment-out the param.

bobr-kun committed:
Some config fixes

bobr-kun committed:
Fix to header file to include <sys/types.h> only for FREE_BSD5 OSes

madf commented:
@bobr-kun I'm ready to merge your request but I see that there are more commits in your https://github.com/bobr-kun/stg/tree/FreeBSD_12.1_build_fix_and_config_updates branch. Please push them so I see that your work there is finished.

bobr-kun commented:
I'd better close this one and open a new one with all the last changes
